lib/types/src/store.rs

```diff
@@ -0,0 +1,72 @@
+use std::collections::HashMap;
+use std::io::Read;
+
+/// Represents a snapshot of parts of the store that mutate
+/// (such as globals and tables)
+#[derive(Debug, Default)]
+pub struct StoreSnapshot {
+    /// Global values at the time the snapshot was taken
+    pub globals: HashMap<u32, u128>,
+}
+
+impl StoreSnapshot {
```

Comment:
I'm a bit unclear on what this is for

Reply:
This is a copy of all the globals in a module.

Used for forking to copy over the global state.

I guess this is in here because the Store doesn't have a public api to access all globals.